### PR TITLE
Refactor filtering

### PR DESCRIPTION
**Pull request by hakhag**
While looking into how we could extend the filtering with other options (like nearest leader), I thought that the current filtering logic was needlessly complex. Before this PR, when a filtering happened, the state of rows was overwritten with the matching rows, and the original state of rows was lost (called `allRows`). To accomodate for this, the prop `allRows` was passed to _alot_ of components which really didn't have anything to do with it, other than having it available when calling `dispatch` (which was also passed around). Also, the reducer-logic was kinda hard to wrap my head around, considering each case called a function which accepted 8 arguments, each either representing the current state or the value of the action passed.

So to make this a little less complex, I've done the following:

1. Removed all the complex reducer-logic and replaced it with a single `useState` object: `filter`. This contains all the current filtering. When for example a search is made, we update `filter.searchTerm` with the corresponding value (with `setFilter`, of course). Note that I considered using a reducer to handle the filtering, but considering it's just one level deep right now, I don't think the extra code required makes it worth it.
2. Instead of having to keep track of the original state by passing `allRows` all over the place, we now do the filtering when the component renders, based on what's in the `filter` object, and store it in a `filteredRows` variable. The value of `filteredRows` is then what's displayed in the table.
3. There was some hardcoded logic in namely `FilterHeader` and `CompetenceFilterInput` based on their type: `'COMPETENCE' | 'MOTIVATION'`. Based on these values, they did different things, like call different dispatches or have different placeholders. I've changed these components to instead have prop functions like `onSelect` and `onThresholdUpdate`, which are called from inside the component. The parentcomponent can then decide what to do when they're called. There was also some hardcoded logic in `SearchInput` which has been replaced in the same manner.

### CODE REVIEW

File: src/components/CompetenceFilterInput.tsx [hunk 1]
```diff
@@ -93,25 +88,14 @@ export default function CompetenceFilterInput({
...
-      filterList: values.map((categoryWithGroup) => categoryWithGroup.category),
-      allRows,
-      searchableColumns,
-    });
+    onSelect(values.map((categoryWithGroup) => categoryWithGroup.category));
   };
 
   return (
     <Autocomplete
-      id={type}
```
Comment (leevi978) [inline src/components/CompetenceFilterInput.tsx:114]: Hvorfor er id fjernet?

File: src/components/FilterHeader.tsx [hunk 2]
```diff
@@ -110,53 +109,29 @@ const RemoveAllTag = (onDelete: { onDelete: () => void }) => {
...
-      type: dispatchThresholdUpdate,
-      threshold,
-      allRows,
-      searchableColumns,
-    });
-  }
-
   function handleThresholdSliderChange(
-    event: React.ChangeEvent<unknown>,
+    _event: React.ChangeEvent<unknown>,
```
Comment (leevi978) [inline src/components/FilterHeader.tsx:127]: Er dette en typo? Ser ikke at parameteren brukes noe sted i funksjonen, så forstår ikke hvorfor navnet skal endres?

File: src/components/FilterHeader.tsx [hunk 3]
```diff
@@ -110,53 +109,29 @@ const RemoveAllTag = (onDelete: { onDelete: () => void }) => {
...
     value: number | number[]
   ) {
     if (Array.isArray(value)) {
       value = value?.[0];
```
Comment (leevi978) [inline src/components/FilterHeader.tsx:131]: Hvorfor er det brukt "?" når det allerede er sjekket om value er et array?  Ifølge if setningen skal value både være definert og være et array, så "?" bør ikke være nødvendig

File: src/components/CompetenceFilterInput.tsx [hunk 4]
```diff
@@ -93,25 +88,14 @@ export default function CompetenceFilterInput({
   );
 
   const handleCategoryChange = (
-    event: ChangeEvent<unknown>,
+    _event: ChangeEvent<unknown>,
```
Comment (leevi978) [inline src/components/CompetenceFilterInput.tsx:91]: Er dette en typo? Ser ikke at parameteren brukes noe sted i funksjonen, så forstår ikke hvorfor navnet skal endres?

File: src/components/CompetenceFilterInput.tsx (hunk 1 above)
Comment (hakhag) [reply src/components/CompetenceFilterInput.tsx:114]: Såvidt jeg kunne se blir `id` kun brukt som en HTML property, altså at elementet får den IDen. Og siden jeg uansett fjernet `type` propen, tenkte jeg denne også kunne fjernes, da den ikke har noen funksjon.

File: src/components/FilterHeader.tsx (hunk 2 above)
Comment (hakhag) [reply src/components/FilterHeader.tsx:127]: Ikke en typo, men jeg har blitt vandt med å prefixe ubrukte parametre med `_`, for å tilfredstille eventuelle lintere.

File: src/components/FilterHeader.tsx (hunk 3 above)
Comment (hakhag) [reply src/components/FilterHeader.tsx:131]: Godt poeng, dog dette er ikke noe jeg har rørt i denne PRen. Jeg kan uansett fikse det.

File: src/components/CompetenceFilterInput.tsx (hunk 1 above)
Comment (leevi978) [reply src/components/CompetenceFilterInput.tsx:114]: Hentet denne beskrivelsen av id feltet fra material-ui sine docs:

> This prop is used to help implement the accessibility logic. If you don't provide this prop. It falls back to a randomly generated id.

Comment (leevi978) [reply src/components/CompetenceFilterInput.tsx:114]: Vet ikke nøyaktig hvilken accessibility logic det er snakk om, eller akkurat hvordan id-propen brukes. Ville bare gjøre deg obs på dette, så kan du selv gjøre en vurdering

**Pull request closed without merging**